provider/machinepool/machine_pool_resource.go

```diff
@@ -681,21 +779,50 @@ func (r *MachinePoolResource) Delete(ctx context.Context, req resource.DeleteReq
 		MachinePool(state.ID.ValueString())
 	_, err := resource.Delete().SendContext(ctx)
 	if err != nil {
-		resp.Diagnostics.AddError(
-			"Cannot delete machine pool",
-			fmt.Sprintf(
-				"Cannot delete machine pool with identifier '%s' for "+
-					"cluster '%s': %v",
-				state.ID.ValueString(), state.Cluster.ValueString(), err,
-			),
-		)
-		return
+		// We can't delete the pool, see if it's the last one:
+		numPools, err2 := r.countPools(ctx, state.Cluster.ValueString())
+		if numPools == 1 && err2 == nil {
+			// It's the last one, issue warning instead of error
+			resp.Diagnostics.AddWarning(
+				"Cannot delete machine pool",
+				fmt.Sprintf(
+					"Cannot delete the last machine pool for cluster '%s'. "+
```

Comment:
May be I am missing something, but should there is check for the scenario:
-I have 2 machinepools: with 1 replica and and default - with 3 replicas (for multiaz cluster)
- I want to delete default machinepool
Will it fail as the last machinepool doesn't fit the cluster requirement?

Reply:
ACK. Thabk you @JohnStrunk